### PR TITLE
Runtime changes common

### PR DESCRIPTION
**Pull request by Eitu33**

### CODE REVIEW

File: massa-grpc/src/api.rs [hunk 1]
```diff
@@ -275,6 +275,8 @@ pub(crate) fn get_largest_stakers(
     // Create the context for the response.
     let context = Some(grpc_api::LargestStakersContext {
         slot: Some(cur_slot.into()),
+        // IMPORTANT TODO: tmp value because testnet_24 and massa proto latest are not synced atm
+        in_downtime: false,
```
Comment (damip) [inline massa-grpc/src/api.rs:279]: what does this do ?

Comment (Eitu33) [reply massa-grpc/src/api.rs:279]: Coming from the proto doc: "If the chain is in downtime, we will return the last snapshot before the downtime"

This is an API type update it has nothing to do with this PR, I just put a filler so we can use our changes.

Comment (damip) [reply massa-grpc/src/api.rs:279]: ok thanks :)

File: massa-execution-exports/Cargo.toml [hunk 2]
```diff
@@ -27,7 +27,7 @@ massa_final_state = { path = "../massa-final-state" }
 massa_ledger_exports = { path = "../massa-ledger-exports", optional = true }
 massa_module_cache = { path = "../massa-module-cache" }
 massa_versioning = { path = "../massa-versioning" }
-massa-sc-runtime = { git = "https://github.com/massalabs/massa-sc-runtime", branch = "main" }
+massa-sc-runtime = { git = "https://github.com/massalabs/massa-sc-runtime", branch = "feature/Improve_ABI_types_in_wasmv1" }
```
Comment (Leo-Besancon) [inline massa-execution-exports/Cargo.toml:30]: /!\ Do not merge until we:
- Merge the massa-sc-runtime common branch into main
- Update the Cargo.toml with a rev of the massa-sc-runtime commit

File: massa-execution-worker/src/interface_impl.rs [hunk 3]
```diff
@@ -603,6 +603,49 @@ impl Interface for InterfaceImpl {
...
+        let public_key = libsecp256k1::PublicKey::parse_slice(
+            public_key_,
+            Some(libsecp256k1::PublicKeyFormat::Full),
+        )?;
+
+        // build the message
+        let prefix = format!("\x19Ethereum Signed Message:\n{}", message_.len());
+        let to_hash = [prefix.as_bytes(), message_].concat();
+        let full_hash = sha3::Keccak256::digest(to_hash);
+        let message = libsecp256k1::Message::parse_slice(&full_hash).unwrap();
```
Comment (damip) [inline massa-execution-worker/src/interface_impl.rs:630]: here we can use expect to give an error message

File: massa-execution-worker/src/interface_impl.rs [hunk 4]
```diff
@@ -603,6 +603,49 @@ impl Interface for InterfaceImpl {
...
+
+        // parse the signature as being (r, s, v)
+        // r is the R.x value of the signature's R point (32 bytes)
+        // s is the signature proof for R.x (32 bytes)
+        // v is a recovery parameter used to ease the signature verification (1 byte)
+        // we ignore the recovery parameter here
+        // see test_evm_verify for an example of its usage
+        let signature = libsecp256k1::Signature::parse_standard_slice(&signature_[..64]).unwrap();
```
Comment (damip) [inline massa-execution-worker/src/interface_impl.rs:638]: the unwrap here can be triggered by an attacker sending an invalid signature

File: massa-execution-worker/src/interface_impl.rs [hunk 5]
```diff
@@ -512,6 +691,48 @@ impl Interface for InterfaceImpl {
...
+    /// # Returns
+    /// A list of keys (keys are byte arrays) that match the given prefix
+    fn get_op_keys_wasmv1(&self, prefix: &[u8]) -> Result<Vec<Vec<u8>>> {
+        // compute prefix range
+        let prefix_range = if !prefix.is_empty() {
+            // compute end of prefix range
+            let mut prefix_end = prefix.to_vec();
+            while let Some(255) = prefix_end.last() {
+                prefix_end.pop();
+            }
```
Comment (Leo-Besancon) [inline massa-execution-worker/src/interface_impl.rs:706]: Note: to update once this is merged: https://github.com/massalabs/massa/pull/4152

File: massa-execution-worker/src/interface_impl.rs (hunk 4 above)
Comment (Eitu33) [reply massa-execution-worker/src/interface_impl.rs]: fixed

File: massa-execution-worker/src/interface_impl.rs (hunk 3 above)
Comment (Eitu33) [reply massa-execution-worker/src/interface_impl.rs]: fixed

File: massa-execution-worker/src/interface_impl.rs (hunk 5 above)
Comment (Leo-Besancon) [reply massa-execution-worker/src/interface_impl.rs]: Done

**Review by Leo-Besancon — APPROVED** [review]

**Pull request merged**